### PR TITLE
Support etag as trailer

### PR DESCRIPTION
**Pull request by akrambek**

### CODE REVIEW

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/ProxyConnectReplyStream.java [hunk 1]
```diff
@@ -557,4 +564,19 @@ private int budgetAvailableWhenProxying(
...
+        final OctetsFW extension = end.extension();
+        if (extension.sizeof() != 0)
+        {
+            final HttpEndExFW httpEndEx = extension.get(streamFactory.httpEndExRO::wrap);
+            ListFW<HttpHeaderFW> trailers = httpEndEx.trailers();
+            String etag = trailers.matchFirst(h -> "etag".equals(h.name().asString())).value().asString();
+            assert etag !=null && !etag.isEmpty();
+            request.etag(etag);
+            request.setEtagInjected(true);
+            this.endExtension = extension;
```
Comment (akrambek) [inline src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/ProxyConnectReplyStream.java]: I am not sure if it is a good idea or not but if I receive etag as the trailer I am marking it as etagEnjected.

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/ProxyAcceptStream.java [hunk 2]
```diff
@@ -170,8 +170,12 @@ private void handlePreferWaitIfNoneMatchRequest(
         short authScope,
         ListFW<HttpHeaderFW> requestHeaders)
     {
-        final String etag = streamFactory.supplyEtag.get();
-
+        String etag = null;
+        HttpHeaderFW etagHeader = requestHeaders.matchFirst(h -> "etag".equals(h.name().asString()));
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/ProxyAcceptStream.java]: Why are we looking for `etag` on the request headers?
Should that be `if-none-match`?

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/ProxyConnectReplyStream.java (hunk 1 above)
Comment (jfallows) [reply src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/ProxyConnectReplyStream.java]: What problem are you attempting to solve by doing this?
For example, what breaks if you don't do this?

**Comment by jfallows** [comment]
Implicitly merged as part of #110

**Pull request closed without merging**